server/sonar-server/src/main/java/org/sonar/server/computation/ComputationService.java

```diff
@@ -44,43 +51,64 @@
   private final DbClient dbClient;
   private final ComputationSteps steps;
   private final ActivityService activityService;
+  private final TempFolder tempFolder;
 
-  public ComputationService(DbClient dbClient, ComputationSteps steps, ActivityService activityService) {
+  public ComputationService(DbClient dbClient, ComputationSteps steps, ActivityService activityService,
+    TempFolder tempFolder) {
     this.dbClient = dbClient;
     this.steps = steps;
     this.activityService = activityService;
+    this.tempFolder = tempFolder;
   }
 
-  public void process(AnalysisReportDto report) {
-    Profiler profiler = Profiler.create(LOG).startInfo(String.format(
-      "#%s - %s - processing analysis report", report.getId(), report.getProjectKey()));
+  public void process(ReportQueue.Item item) {
+    Profiler profiler = Profiler.create(LOG).startDebug(String.format(
+      "Analysis of project %s (report %d)", item.dto.getProjectKey(), item.dto.getId()));
 
-    ComponentDto project = loadProject(report);
+    ComponentDto project = loadProject(item);
     try {
-      ComputationContext context = new ComputationContext(report, project);
+      File reportDir = extractReportInDir(item);
+      BatchOutputReader reader = new BatchOutputReader(reportDir);
+      ComputationContext context = new ComputationContext(reader, project);
       for (ComputationStep step : steps.orderedSteps()) {
         if (ArrayUtils.contains(step.supportedProjectQualifiers(), context.getProject().qualifier())) {
-          Profiler stepProfiler = Profiler.create(LOG).startInfo(step.getDescription());
+          Profiler stepProfiler = Profiler.createIfDebug(LOG).startDebug(step.getDescription());
           step.execute(context);
-          stepProfiler.stopInfo();
+          stepProfiler.stopDebug();
         }
       }
-      report.succeed();
+      item.dto.succeed();
 
     } catch (Exception e) {
-      report.fail();
+      item.dto.fail();
       throw Throwables.propagate(e);
 
     } finally {
-      logActivity(report, project);
+      logActivity(item.dto, project);
```

Comment:
set the finished at date here instead of ReportQueue#remove, as you rightfully commented